src/Toast.php

```diff
@@ -0,0 +1,398 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Yii\Bootstrap5;
+
+use JsonException;
+use Yiisoft\Arrays\ArrayHelper;
+use Yiisoft\Html\Html;
+
+use function array_merge;
+
+/**
+ * Toasts renders a toast bootstrap widget.
+ *
+ * For example,
+ *
+ * ```php
+  * echo Toast::widget()
+ *     ->title('Hello world!')
+ *     ->dateTime('a minute ago')
+ *     ->body('Say hello...')
+ *     ->begin();
+ * ```
+ *
+ * The following example will show the content enclosed between the {@see begin()}
+ * and {@see end()} calls within the toast box:
+ *
+ * ```php
+ * echo Toast::widget()
+ *     ->title('Hello world!')
+ *     ->dateTime('a minute ago')
+ *     ->begin();
+ *
+ * echo 'Say hello...';
+ *
+ * echo Toast::end();
+ * ```
+ *
+ * @see https://getbootstrap.com/docs/5.0/components/toasts/
+ */
+final class Toast extends Widget
+{
+    /**
+     * @var string the body content in the alert component. Note that anything between
+     * the {@see begin()} and {@see end()} calls of the Toast widget will also be treated
+     * as the body content, and will be rendered before this.
+     */
+    private string $body = '';
+    /**
+     * @var string The title content in the toast.
+     */
+    private string $title = '';
+    /**
+     * @var string The footer content in the toast.
+     */
+    private string $footer = '';
+    /**
+     * @var string The date time the toast message references to.
+     * This will be formatted as relative time (via formatter component). It will be omitted if
+     * set to `false` (default).
+     */
+    private string $dateTime = '';
+    /**
+     * @var array the options for rendering the close button tag.
+     * The close button is displayed in the header of the toast. Clicking on the button will hide the toast.
+     *
+     * The following special options are supported:
+     *
+     * - tag: string, the tag name of the button. Defaults to 'button'.
+     * - label: string, the label of the button. Defaults to '&times;'.
+     *
+     * The rest of the options will be rendered as the HTML attributes of the button tag.
+     * Please refer to the [Toast documentation](https://getbootstrap.com/docs/5.0/components/toasts/)
+     * for the supported HTML attributes.
+     */
+    private array $closeButton = [];
+    /**
+     * @var array additional title options
+     *
+     * The following special options are supported:
+     *
+     * - tag: string, the tag name of the button. Defaults to 'strong'.
+     *
+     * @see Html::renderTagAttributes() for details on how attributes are being rendered.
+     */
+    public array $titleOptions = [];
```

Comment:
Should it be public? I don't think so.

Reply:
https://github.com/yiisoft/yii-bootstrap5/commit/d0abc296dd4c5bdce784d9469d288f6969ce97b8

Reply:
I'm not sure. Basically this is just an edited version of https://github.com/yiisoft/yii-bootstrap5/blob/master/src/Modal.php

Reply:
There it wasn't public. Also I've made some fixes: https://github.com/yiisoft/yii-bootstrap5/commit/a2749520a40cce86a1a98a64867acdea7b829d83 Likely protected methods aren't correct in Modal as well.